astropy/coordinates/tests/test_arrays.py

```diff
@@ -41,10 +42,16 @@ def test_angle_arrays():
     npt.assert_almost_equal(a6.value, 945.0)
     assert a6.unit is u.degree
 
-    with pytest.raises(TypeError):
+    with ExitStack() as stack:
+        stack.enter_context(pytest.raises(TypeError))
         # Arrays where the elements are Angle objects are not supported -- it's
         # really tricky to do correctly, if at all, due to the possibility of
         # nesting.
+        if not NUMPY_LT_1_18:
+            stack.enter_context(
+                pytest.warns(DeprecationWarning,
+                             match='automatic object dtype is deprecated'))
```

Comment:
Shouldn't we aim to make sure the code below doesn't emit a warning?

Reply:
Oh wait that's due to https://github.com/numpy/numpy/issues/15041 right?

Reply:
Yes, numpy emits the warning. Since this is just a test that something fails, I think it is not all that important whether there is a warning as well. Eventually, the warning might just change to another exception, but that is OK too.